database/database.go

```diff
@@ -237,3 +251,76 @@ func connectionString(creds DBCredentials) string {
 		mongoWriteConcernTimeout,
 	)
 }
+
+// ensureDBSchema checks that we have all collections and indexes we need and
+// creates them if needed.
+// See https://docs.mongodb.com/manual/indexes/
+// See https://docs.mongodb.com/manual/core/index-unique/
+func ensureDBSchema(ctx context.Context, db *mongo.Database) error {
+	// schema defines a mapping between a collection name and the indexes that
+	// must exist for that collection.
+	schema := map[string][]mongo.IndexModel{
+		dbUsersCollection: {
+			{
+				Keys:    bson.D{{"sub", 1}},
+				Options: options.Index().SetName("usersSubUnique").SetUnique(true),
+			},
+		},
+		dbSkylinksCollection: {
+			{
+				Keys:    bson.D{{"skylink", 1}},
+				Options: options.Index().SetName("skylinksSkylinkUnique").SetUnique(true),
+			},
+		},
+		dbUploadsCollection: {
+			{
+				Keys:    bson.D{{"user_id", 1}},
+				Options: options.Index().SetName("uploadsUserID"),
+			},
+			{
+				Keys:    bson.D{{"skylink_id", 1}},
+				Options: options.Index().SetName("uploadsSkylinkID"),
+			},
+		},
+		dbDownloadsCollection: {
+			{
+				Keys:    bson.D{{"user_id", 1}},
+				Options: options.Index().SetName("uploadsUserID"),
```

Comment:
```suggestion
				Options: options.Index().SetName("downloadsUserID"),
```

(x2)

Reply:
Copy-paste strikes again! No wonder, tbh, all the uploads/downloads code is identical. I actually considered writing it once and calling it with a parameter but it would have turned out more complicated that the current form. :)